back-end/helpers/intent_analyzer.js

```diff
@@ -0,0 +1,56 @@
+export default class IntentAnalyzer{
+    /* What it should result in:
+    [
+        {
+          question: "What is something that Waluigi and Wario often say?",
+          children: {
+            "The brothers Waluigi and Wario have their catchphrase 'waaa'.": 100,
+          }
+        },
+        {
+          question: "Where should I go for a good time and great learning experience?",
+          children: {
+            "You should go to the University of Toronto.": 10,
+            "You should go to the Technology Leadership Initiative.": 90
+          }
+        }, 
+        {
+          question: "Hey BestFlow, what time is it?",
+          children: {
+            "It's time for you to get a watch, would you like to take a look at our product?": 10,
+            "Showtime showtime what I'm John Laurens in the place to be.": 20,
+            "Summertime! School's out, scream and shout.": 70
+          }
+        }
+    ]
+      */
+
+    static async formatIntents(intents) { // Formats the mongoose Model into desired array for IntentLister to use
+      if (Object.keys(intents).length == 0) {
```

Comment:
I don't think this if block is necessary since if there are no intents, then your code below with analyzedIntents will equal `{ }` anyway.